src/header/map.rs

```diff
@@ -135,7 +135,9 @@ pub struct ValuesMut<'a, T: 'a> {
 #[derive(Debug)]
 pub struct Drain<'a, T: 'a> {
     idx: usize,
-    map: *mut HeaderMap<T>,
+    len: usize,
+    entries: *mut [Bucket<T>],
+    extra_values: *mut Vec<ExtraValue<T>>,
```

Comment:
AFAICT, these are never null & are not null-checked; consider using `NonNull` to encode this in the type system.

Reply:
We could, though I'm not sure it gives much benefits here... It's mostly a layout optimization. but is otherwise more annoying to use.

Reply:
The motivation I had in mind was mostly "making it clearer to a future reader that these are never null", rather than optimizations. Up to you.